InvocationTargetException when running sanity-checks on 3.0.0-RC2

### Bug description

The steps to reproduce

1. Checkout on `3.0.0-RC2`
2. Run `./mvnw -f extension/ldap/pom.xml`
3. It will be failling with
```
[ERROR] Failed to execute goal org.codehaus.gmavenplus:gmavenplus-plugin:3.0.0:execute (sanity-checks) on project camel-quarkus-ldap: Error occurred while calling a method on a Groovy class from classpath.: InvocationTargetException: For input string: "0-RC2" -> [Help 1]
```

It needs to strip anything after "-" in `compareVersion` method in `tooling/scripts/sanity-checks.groovy`

--- a/tooling/scripts/sanity-checks.groovy
+++ b/tooling/scripts/sanity-checks.groovy
@@ -51,7 +51,8 @@ if (!badDeps.isEmpty()) {
 }
 
 // check jvmSince and nativeCheck not newer than current SNAPSHOT version
-final String currentVersion = project.version - '-SNAPSHOT'
+def versionRegex = ~/\-.*/
+final String currentVersion = project.version - versionRegex
 final String jvmSince    = project.properties['camel.quarkus.jvmSince']
 final String nativeSince = project.properties['camel.quarkus.nativeSince']
 

